lib/ansible/plugins/connection/kubernetes.py

```diff
@@ -0,0 +1,328 @@
+                            self._play_context.password and
+                            self._play_context.kubernetes_port):
+
+                        self.namespace = self._play_context.kubernetes_namespace
+                        kubernetes_config = self.generate_kubernetes_config()
+                        active_context = self.requested_context()
+                        kube_loader = KubeConfigLoader(config_dict=kubernetes_config,
+                                                       active_context=active_context)
+                        kube_loader.load_and_set()
+                    elif self._play_context.kubernetes_config_file:
+                        config.load_kube_config(config_file=self._play_context.kubernetes_config_file)
+                        self.namespace = self._play_context.kubernetes_namespace
+                    else:
+                        config.load_kube_config()
+                        _, self.active_context = config.list_kube_config_contexts()
+                        self.namespace = self.active_context['context']['namespace']
+
+                    self.api = core_v1_api.CoreV1Api()
+                    self._connected = True
+                    display.vvv(u"ESTABLISH CONNECTION FOR USER: {0}".format(self._play_context.remote_user),
+                                host=self._play_context.remote_addr)
+        except ConfigException as e:
+            AnsibleError("Kubernetes Connection Config Error: {0}".format(e.message))
+        except ApiException as e:
+            AnsibleError("Kubernetes Connection API Error: {0}".format(e.message))
+        except Exception as e:
+            AnsibleError("Unknown Error: {0}".format(e.message))
+        return self
+
+    def exec_command(self, cmd, in_data=None, sudoable=False):
+        ''' run a command on a kubernetes pod'''
+        super(Connection, self).exec_command(cmd, in_data=in_data, sudoable=sudoable)
+        stdout = ''
+        stderr = ''
+        returncode = 0
+
+        local_cmd = [self._play_context.executable, '-c', cmd]
+        display.vvv("EXEC {0}".format(to_text(local_cmd)), host=self._play_context.remote_addr)
+        local_cmd = [to_bytes(i, errors='surrogate_or_strict') for i in local_cmd]
+        try:
+            self.response = self.api.connect_get_namespaced_pod_exec(name=self.name,
+                                                                     namespace=self.namespace,
+                                                                     command=local_cmd,
+                                                                     stderr=True,
+                                                                     stdin=True,
+                                                                     stdout=True,
+                                                                     tty=False,
+                                                                     _preload_content=False)
```

Comment:
@chrrrles are you asking about the method or the `_preload_content` argument?  If the argument is not provided this doesn't work correctly.